modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v5_24_0/CPConfigurationUpgradeProcess.java

```diff
@@ -0,0 +1,187 @@
+import com.liferay.portal.kernel.service.ClassNameLocalServiceUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.CalendarFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * @author Matyas Wollner
+ */
+public class CPConfigurationUpgradeProcess extends UpgradeProcess {
+
+	public CPConfigurationUpgradeProcess(
+		ClassNameLocalService classNameLocalService,
+		CPConfigurationEntryLocalService cpConfigurationEntryLocalService,
+		CPConfigurationListLocalService cpConfigurationListLocalService,
+		Language language) {
+
+		_classNameLocalService = classNameLocalService;
+		_cpConfigurationEntryLocalService = cpConfigurationEntryLocalService;
+		_cpConfigurationListLocalService = cpConfigurationListLocalService;
+		_language = language;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement configurationListPreparedStatement =
+				connection.prepareStatement(
+					StringBundler.concat(
+						"select CommerceCatalog.name, CommerceCatalog.userId, ",
+						"CommerceCatalog.catalogDefaultLanguageId, ",
+						"Group_.groupId, ",
+						"CPConfigurationList.CpConfigurationListId from ",
+						"CommerceCatalog join Group_ on ",
+						"CommerceCatalog.commerceCatalogId=group_.classPK and ",
+						"Group_.classNameId=? left join CPConfigurationList ",
+						"on Group_.groupId=CPConfigurationList.groupId and ",
+						"masterCPConfigurationList= ?"));
+			PreparedStatement configurationEntryPreparedStatement =
+				connection.prepareStatement(
+					StringBundler.concat(
+						"select CPDefinition.CPDefinitionId, ",
+						"CPDefinitionInventory.allowedOrderQuantities, ",
+						"CPDefinitionInventory.backOrders, ",
+						"CpDefinitionInventory.CPDefinitionInventoryEngine, ",
+						"CpDefinitionInventory.displayAvailability, ",
+						"CPDefinitionInventory.displayStockQuantity, ",
+						"CpDefinitionInventory.lowStockActivity, ",
+						"CpDefinitionInventory.maxOrderQuantity, ",
+						"CpDefinitionInventory.minOrderQuantity, ",
+						"CpDefinitionInventory.minStockQuantity, ",
+						"CpDefinitionInventory.multipleOrderQuantity, ",
+						"CPDAvailabilityEstimate.",
+						"commerceAvailabilityEstimateId from CPDefinition ",
+						"join CpDefinitionInventory on ",
+						"CPDefinition.CPDefinitionId=",
+						"CpDefinitionInventory.CPDefinitionId left join ",
+						"CPConfigurationEntry on CPConfigurationEntry.classPK=",
+						"CPDefinition.CPDefinitionId and ",
+						"CPConfigurationEntry.CPConfigurationListId=? left ",
+						"join CPDAvailabilityEstimate on ",
+						"CPDefinition.CProductId=",
+						"CPDAvailabilityEstimate.CProductId where ",
+						"CPDefinition.groupId=? and ",
+						"CPConfigurationEntry.CPConfigurationListId is ",
```

Comment:
Table joins could also be sorted based on their names (since they do not depend on each other), not sure if this is required though.